android/java/org/chromium/chrome/browser/rewards/RewardsCustomTabRootUiCoordinator.java

```diff
@@ -0,0 +1,134 @@
+            @NonNull Supplier<TabCreatorManager> tabCreatorManagerSupplier,
+            @NonNull FullscreenManager fullscreenManager,
+            @NonNull ObservableSupplier<CompositorViewHolder> compositorViewHolderSupplier,
+            @NonNull Supplier<TabContentManager> tabContentManagerSupplier,
+            @NonNull Supplier<SnackbarManager> snackbarManagerSupplier,
+            @NonNull ObservableSupplierImpl<EdgeToEdgeController> edgeToEdgeControllerSupplier,
+            @ActivityType int activityType,
+            @NonNull Supplier<Boolean> isInOverviewModeSupplier,
+            @NonNull AppMenuDelegate appMenuDelegate,
+            @NonNull StatusBarColorProvider statusBarColorProvider,
+            @NonNull IntentRequestTracker intentRequestTracker,
+            @NonNull Supplier<CustomTabToolbarCoordinator> customTabToolbarCoordinator,
+            @NonNull Supplier<BrowserServicesIntentDataProvider> intentDataProvider,
+            @NonNull BackPressManager backPressManager,
+            @NonNull Supplier<CustomTabActivityTabController> tabController,
+            @NonNull Supplier<CustomTabMinimizeDelegate> minimizeDelegateSupplier,
+            @NonNull Supplier<CustomTabFeatureOverridesManager> featureOverridesManagerSupplier,
+            @Nullable View baseChromeLayout,
+            @NonNull EdgeToEdgeManager edgeToEdgeManager) {
+        super(
+                activity,
+                shareDelegateSupplier,
+                tabProvider,
+                profileSupplier,
+                bookmarkModelSupplier,
+                tabBookmarkerSupplier,
+                tabModelSelectorSupplier,
+                browserControlsManager,
+                windowAndroid,
+                activityLifecycleDispatcher,
+                layoutManagerSupplier,
+                menuOrKeyboardActionController,
+                activityThemeColorSupplier,
+                modalDialogManagerSupplier,
+                appMenuBlocker,
+                supportsAppMenuSupplier,
+                supportsFindInPage,
+                tabCreatorManagerSupplier,
+                fullscreenManager,
+                compositorViewHolderSupplier,
+                tabContentManagerSupplier,
+                snackbarManagerSupplier,
+                edgeToEdgeControllerSupplier,
+                activityType,
+                isInOverviewModeSupplier,
+                appMenuDelegate,
+                statusBarColorProvider,
+                intentRequestTracker,
+                customTabToolbarCoordinator,
+                intentDataProvider,
+                backPressManager,
+                tabController,
+                minimizeDelegateSupplier,
+                featureOverridesManagerSupplier,
+                baseChromeLayout,
+                edgeToEdgeManager);
+    }
+
+    @Override
+    public int getControlContainerHeightResource() {
+        return R.dimen.custom_tabs_control_container_rewards_height;
```

Comment:
you're overriding this to make the height zero?

Reply:
yes exactly. i am just overriding BaseCustomTabActivity and i needed to make bytecode changes so that we can override the height. we want to have the fullscreen ui. and without overriding the height, it would add toolbar space on the top.

Reply:
First of all you need to explain why you are doing things in comments. No one else is going to have any idea what is going on here. Second, if we need a fullscreen ui capability that doesn't currently exist, we shouldn't be creating a rewards specific subclass to handle that. If this was `FullscreenCustomTabRootUiCoordinator` it would be much easier to understand. Third, why is this fundamentally different from other custom tab activities? Isn't there already fullscreen functionality for something like this? It still seems like we're misusing this.

Reply:
@bridiver basically i took https://github.com/brave/brave-core/blob/2867ea8e086daf85e925457ce0735ffb1932edd3/android/java/org/chromium/chrome/browser/brave_leo/BraveLeoCustomTabRootUiCoordinator.java as ref which was added for Leo. and it was removed here : https://github.com/brave/brave-core/pull/26823/files#diff-0ef300181a2f5bce1c1aaeb81180379d85765d419d55574803948a557f277ead
i wanted to create an independent class but rewards is the only usecase we have right now.

Reply:
in the custom tab activities we have, we have a top margin to include url bar.

Reply:
and before CR133 we didn't need to use bytecode for anything. but after the rebase i had to update the changes : https://github.com/brave/brave-core/pull/26945/commits/35520597166adbd729df8c44bb1134e78b2bd150

Reply:
now I'm more confused because in the other comment it says this is supposed to open in a regular tab, but that's exactly what the PR you referenced above removes this code to do. Part of the problem here is that none of these issues have any kind of useful information, but as I said the code also needs comments explaining what is going on. I also see other fullscreen UI in upstream that doesn't do this. What urls are we opening in fullscreen like this?

Reply:
we open URLs mainly for finishing the auth flow for any custodian accounts. on `Connect` click, we mainly need to close the full screen panel and open the url in new tab.